spec/support/capybara.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.configure do |config|
+  headless = ENV.fetch('HEADLESS', true) != 'false'
+
+  config.before(:each, type: :system) do
+    driven_by :rack_test
+  end
+
+  config.before :each, type: :system, js: true do
+    # Docker environment
+    url = if headless
+            "http://#{ENV['SELENIUM_REMOTE_HOST']}:4444/wd/hub"
+          else
+            'http://host.docker.internal:9515'
+          end
+
+    driven_by :selenium, using: :chrome, options: {
+      browser:              :remote,
+      url:                  url,
+      desired_capabilities: :chrome
+    }
+
+    # Find Docker IP address
+    Capybara.server_host = if headless
+                             `/sbin/ip route|awk '/scope/ { print $9 }'`.strip
+                           else
+                             '0.0.0.0'
+                           end
+    Capybara.server_port = '43447'
+    Capybara.app_host    = "http://#{Capybara.current_session.server.host}:#{Capybara.current_session.server.port}"
+  end
+
+  config.after :each, type: :system, js: true do
+    page.driver.browser.manage.logs.get(:browser).each do |log|
```

Comment:
This is interesting, what brought this about?

Reply:
It was in Noel Rappin’s blog post for using system tests and it seems sane to us so we used it.